precompilation/precompile_PrettyTables.jl

```diff
@@ -1,164 +1,162 @@
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Matrix{Float32},String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003888151
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Vector{Float64},String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003883694
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Vector{UInt64},String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003880366
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Matrix{Char},String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003868918
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Matrix{UInt32},String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003866396
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,UInt8,String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003862394
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,String,String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003855481
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Matrix{Float16},String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003847007
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Symbol,String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003842512
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Int16,String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003836524
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Int8,String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003821785
-    Base.precompile(Tuple{typeof(_process_data_cell_text),ProcessedTable,Int64,String,Int64,Int64,Int64,Int64,Crayon,Symbol,Ref{Any}})   # time: 0.003797233
-    Base.precompile(Tuple{Core.kwftype(typeof(_parse_cell_text)),NamedTuple{(:autowrap, :cell_data_type, :cell_first_line_only, :column_width, :compact_printing, :has_color, :limit_printing, :linebreaks, :renderer), Tuple{Bool, DataType, Bool, Int64, Bool, Bool, Bool, Bool, Val{:print}}},typeof(_parse_cell_text),UrlTextCell})   # time: 0.003454054
-    Base.precompile(Tuple{typeof(_print_custom_text_cell!),Display,UrlTextCell,String,Crayon,Int64,Ref{Any}})   # time: 0.001846952
-    Base.precompile(Tuple{Core.kwftype(typeof(_parse_cell_html)),NamedTuple{(:allow_html_in_cells, :cell_first_line_only, :compact_printing, :limit_printing, :linebreaks, :renderer), Tuple{Bool, Bool, Bool, Bool, Bool, Val{:print}}},typeof(_parse_cell_html),String})   # time: 0.001669777
-    Base.precompile(Tuple{typeof(_print_custom_text_cell!),Display,AnsiTextCell,String,Crayon,Int64,Ref{Any}})   # time: 0.001618891
+    Base.precompile(Tuple{typeof(_pt_text),IOContext{Base.TTY},PrintInfo})   # time: 0.22947502
```

Comment:
You could check if using SnoopPrecompile.jl instead would not reduce the amount of code needed. See e.g. https://github.com/JuliaData/DataFrames.jl/blob/main/src/other/precompile.jl

(this is an independent thing from this PR)

Reply:
Thanks! I did not know about SnoopPrecompile until now :D Only SnoopCompile